### PR TITLE
Document pragma(mangle) special case on OSX

### PR DESCRIPTION
**Pull request by Geod24**

### CODE REVIEW

File: spec/pragma.dd [hunk 1]
```diff
@@ -144,6 +144,11 @@ $(H3 $(LNAME2 mangle, $(D pragma mangle)))
     $(P There must be one $(ASSIGNEXPRESSION) and it must evaluate at compile time to a string literal.
     )
 
+    $(IMPLEMENTATION_DEFINED On Mac OSX, an extra underscore (`_`) is prepended to the string since 2.079,
```
Comment (jacob-carlborg) [inline spec/pragma.dd]: Not sure what the rest of the documentation says but the official name by Apple is `macOS`.

**Comment by Geod24** [comment]
Looks like this would also happen on Win32 ( https://github.com/dlang/dmd/pull/7620 ) ?
CC @ibuclaw

File: spec/pragma.dd [hunk 2]
```diff
@@ -144,6 +144,11 @@ $(H3 $(LNAME2 mangle, $(D pragma mangle)))
...
+        as is done by the C/C++ toolchain. This allows to use the same `pragma(mangle)` for all POSIX
```
Comment (JinShil) [inline spec/pragma.dd]: "allows to use" --> "allows using"

**Comment by ibuclaw** [comment]
@Geod24 yes that's correct IIRC.

**Comment by Geod24** [comment]
Fixed:
- Added win32 and rephrased the last sentence
- `Mac OSX` => `macOS`
- `allows to use` => `allows using`

**Pull request merged**